### PR TITLE
Replace Wandb callback's finalize with no-op

### LINKED ISSUE #1116: Wandb logger doesn't upload saved model checkpoint for final epoch
## 🐛 Bug

When training a model on the TPU and using the wandb logger, the checkpoint for the last epoch trained doesn't get uploaded to wandb.

### To Reproduce

Colab notebook: https://colab.research.google.com/drive/1oPaRWGZcz6YEol012xFADN42LV-jowtT

### PR DESCRIPTION
**Pull request by bkkaggle**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1116 and #1220 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -108,6 +108,7 @@ def finalize(self, status: str = 'success') -> None:
             wandb.join(exit_code)
         except TypeError:
             wandb.join()
+    return exit_code
```
Comment (Borda) [inline pytorch_lightning/loggers/wandb.py]:
```suggestion
        return exit_code
```

**Comment by Borda** [comment]
@bkkaggle could you pls add a note to changelog?

**Comment by borisdayma** [comment]
The change to the callback looks good to me.

**Comment by jeremyjordan** [comment]
@bkkaggle can you update the corresponding tests? can't merge with failing tests

**Comment by Borda** [comment]
@calclavia @borisdayma can the failing platform-dependent?
it seems that this fix passes on Win but fails on Linux/Unix

**Comment by borisdayma** [comment]
I'm actually surprised. Right now wandb.join is not called anymore and finalize does not do anything.
The test actually just needs to be removed (or we need to check the return value is just 1).

**Comment by Borda** [comment]
@bkkaggle could you pls implement https://github.com/PyTorchLightning/pytorch-lightning/pull/1193#issuecomment-604498805 suggestion
@borisdayma what do you recommend remove or check return value?

**Comment by borisdayma** [comment]
I would just remove finalize method (not sure if pytorch-lightning checks the return value).

If @bkkaggle doesn't have the time I can update the PR.

**Comment by borisdayma** [comment]
For info, here are the changes I suggest: https://github.com/PyTorchLightning/pytorch-lightning/compare/master...borisdayma:fix-wandb

**Comment by borisdayma** [comment]
I am currently testing proposed changes with a W&B sweep and it seems to be working well: [Sweep run](https://app.wandb.ai/borisd13/lightning-kitti/sweeps/lrqc8fu3/workspace?workspace=user-borisd13)

I also had to change the `watch` method as it could not be called at the moment.

I adapted the semantic segmentation example to customize unet, gradient accumulation, etc. Still discovering pytorch-lightning features. Really impressive!
Only thing I cannot test right now is multiple GPU's…

**Comment by Borda** [comment]
@borisdayma minds send update PR for semantic Segmentation?

**Comment by borisdayma** [comment]
Looks good to me! I'll make separate one to update the `watch` method and avoid conflicts.

**Comment by bkkaggle** [comment]
@borda I removed the `finalize()` method and its tests and updated the changelog.

**Comment by Borda** [comment]
it seems that there is issue with examples, could you please rebase `master`
